test/functional/VM_Test/src/j9vm/test/booleans/BooleanTest.java

```diff
@@ -0,0 +1,287 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package j9vm.test.booleans;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Field;
+
+import sun.misc.Unsafe;
+
+
+public class BooleanTest {
+	
+	static native boolean setUp();
+	static native boolean tearDown();
+	static native void putInstanceBoolean(Class<?> clazz, Object instance, String fieldName, String fieldSig, int value);
+	static native void putStaticBoolean(Class<?> clazz, String fieldName, String fieldSig, int value);
+	static native int getInstanceBoolean(Class<?> clazz, Object instance, String fieldName, String fieldSig);
+	static native int getStaticBoolean(Class<?> clazz, String fieldName, String fieldSig);
+	public static native boolean returnBooleanValue(int val);
+	
+	public static class TestBool {
+		public boolean z;
+		public static boolean zStatic;
+	}
+	
+	
+	static Unsafe unsafe;
+
+	static boolean testSuccess = true;
+	static int failedTest = 0;
+	static {
+		System.loadLibrary("j9ben");
+		
+		try {
+			Field f = sun.misc.Unsafe.class.getDeclaredField("theUnsafe");
+			f.setAccessible(true);
+			unsafe = (sun.misc.Unsafe) f.get(null);
+		} catch (NoSuchFieldException | SecurityException | IllegalArgumentException | IllegalAccessException e) {
+			e.printStackTrace();
```

Comment:
This should do a `throw new RuntimeException(e)` or call a `fail()` method to indicate these tests can't be run